### DISCUSSION TITLE
[pie chart] Add an option to sort the labels by a metric not used in chart

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18527] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently the labels in a Pie chart are sorted either alphabetically(in ascending order) or can be sorted by the metric value used in the chart. But there are usecases, where we would like to sort it by another metric which is not used in the chart.
An example would be pipeline stages, should be sorted by stage id which is not in stage label or metric.

![image](https://user-images.githubusercontent.com/74634977/140842791-3d0cd155-3578-4da4-ad5f-03fa2bce0516.png)

**Describe the solution you'd like**
When the user selects 'Sort by Metric' option, by default it should use the metric used in the chart. But it should also allow to change the sorting metric and the sorting order(ascending/descending).

**Describe alternatives you've considered**

**Additional context**

**Comment by junlincc** [comment]
@mayurnewase would you like to take a look at this one? don't forget to setup your committer right in github so we can start collaborating on community issues :) 

cc @villebro
